projects/ADuCM3029_demo_cn0428_cn0429/src/ADuCM3029_demo_cn0428_cn0429.cpp

```diff
@@ -0,0 +1,1307 @@
+			eSensorResult = pGasSensor->openWithAddr(readChar);
+
+			memset(cmdResponseBuff, 0, sizeof(cmdResponseBuff));
+			eSensorResult = pGasSensor->Read200RCal(cmdResponseBuff);
+			rcal = cmdResponseBuff[0];
+
+			flushBuff(TXbuff, sizeof(TXbuff));
+			eSensorResult = pGasSensor->RunPulseTest(pulseAmplitude, pulseDuration);
+			if (eSensorResult != SENSOR_ERROR_NONE) UART_TX("ERROR!" _EOS);
+			else {
+				sprintf((char*)TXbuff,"Pulse test started%s", _EOS);
+				UART_TX((const char*)TXbuff);
+			}
+			pGasSensor->close();
+		}
+
+		if (check_string((uint8_t *)"PULSE_RESULT", 12)) {
+			readChar = GetChar();
+			pulseAmplitude = GetChar();	// read pulseAmplitude as hex
+			pulseDuration = GetChar();	// read pulseDuration as hex
+			DeleteChar(3);
+			eSensorResult = pGasSensor->openWithAddr(readChar);
+			flushBuff(TXbuff, sizeof(TXbuff));
+			eSensorResult = pGasSensor->ReadPulseTestResults(&pulseResultBuffer[0],
+					pulseAmplitude, pulseDuration);
+			if (eSensorResult != SENSOR_ERROR_NONE) UART_TX("ERROR!" _EOS);
+			else {
+				UART_TX("time [msec], current [uA]" _EOS);
+				uint16_t valToSend = (((pulseDuration + 10) * 1000) / 110);
+				uint16_t cnt = 0;
+				uint16_t cnt2 = 1;
+				while (cnt < valToSend) {
+					flushBuff(TXbuff, sizeof(TXbuff));
+					unsigVal = ((pulseResultBuffer[cnt2] << 8) | pulseResultBuffer[cnt2+1]);
+					sprintf((char*)TXbuff,"%d, %.2f%s", ((cnt)*110),
+						((32768.0 - unsigVal)/65535) * ((1835 * 2 * 1000)/rcal), _EOS);
+					UART_TX((const char*)TXbuff);
+					cnt++;
+					cnt2 += 2;
+				}
+			}
+			pGasSensor->close();
+		}
+
+		if (check_string((uint8_t *)"READALLSENSORS", 14)) {
+			flushBuff(TXbuff, sizeof(TXbuff));
+			if (S1_detected) strcat((char*)TXbuff, "Sensor 1[ppb], ");
+			if (S2_detected) strcat((char*)TXbuff, "Sensor 2[ppb], ");
+			if (S3_detected) strcat((char*)TXbuff, "Sensor 3[ppb], ");
+			if (S4_detected) strcat((char*)TXbuff, "Sensor 4[ppb], ");
+			UART_TX((const char*)TXbuff);
+			FSM_State = STREAM_GAS;
+		}
+		break;
+
+	case STREAM_GAS:
+		flushBuff(TXbuff, sizeof(TXbuff));
+		if (S1_detected) {
+			flushBuff(gBuff, sizeof(gBuff));
+			sprintf((char*)gBuff, "%ld, ", SensorReadoutPPB(ADI_CFG_I2C_SENSOR1_ADDR));
+			strcat((char*)TXbuff, (char*)gBuff);
+		}
+		if (S2_detected) {
+			flushBuff(gBuff, sizeof(gBuff));
+			sprintf((char*)gBuff, "%ld, ", SensorReadoutPPB(ADI_CFG_I2C_SENSOR2_ADDR));
+			strcat((char*)TXbuff, (char*)gBuff);
+		}
+		if (S3_detected) {
+			flushBuff(gBuff, sizeof(gBuff));
+			sprintf((char*)gBuff, "%ld, ", SensorReadoutPPB(ADI_CFG_I2C_SENSOR3_ADDR));
+			strcat((char*)TXbuff, (char*)gBuff);
+		}
+		if (S4_detected) {
+			flushBuff(gBuff, sizeof(gBuff));
+			sprintf((char*)gBuff, "%ld, ", SensorReadoutPPB(ADI_CFG_I2C_SENSOR4_ADDR));
+			strcat((char*)TXbuff, (char*)gBuff);
+		}
+		UART_TX((const char*)TXbuff);
+		break;
+
+	case WATER: {
+		if(initialcycle) {
+			//Do initial read of water quality board transmit buffer
+			if (S1_detected) {
```

Comment:
similarly, related to comment https://github.com/analogdevicesinc/EVAL-ADICUP3029/pull/12/commits/4f5694854b122faead93c95e10cae42d6207dad4#r243179682  ; this can be a `for` block

Reply:
This one might not save much space. I would have to also add a sprintf, break, and a flag and comparison for if it exited the loop without breaking. I would prefer to leave as-is.